File: frontend/src/service/CommentService.js
class CommentService{
    commentsURL = "/api/v1/posts/{postId}/comments";

    getCommentsByPostId(postId, pageParams = {}){
        console.log('Fetching comments for post ', postId);
        const {page = 0, size = 5} = pageParams;
        return fetch(this.commentsURL.replace('{postId}', postId) + '?' + new URLSearchParams({
            page, size
        }),
        {
            headers: {
              'Accept': 'application/json'
            }
        }
        );
    }

    publishComment(postId, comment){
        console.log('Publishing comment for post ', postId, comment);
        return fetch(this.commentsURL.replace('{postId}', postId),
        {
            method: 'POST',
            headers: {
              'Accept': 'application/json',
              'Content-Type': 'application/json'
            },
            body: JSON.stringify(comment)
        }
        );
    }

    deleteComment(postId, commentId){
        return fetch(this.commentsURL.replace('{postId}', postId) + `/${commentId}`,
        {
            method: 'DELETE',
        }
        );
    }
}

export default new CommentService;

File: src/main/java/com/wz/postcommentdemo/service/CommentService.java
package com.wz.postcommentdemo.service;

import com.wz.postcommentdemo.entity.Comment;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface CommentService {
    Comment addComment(Long postId, Comment comment);
    Page<Comment> getComments(Long postId, Pageable pageable);

    void deleteComment(Long postId, Long commentId);
}
